src/tokenizer/sqlite.cpp

```diff
@@ -0,0 +1,222 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#include "tokenizer/sqlite.hpp"
+#include "utils.hpp"
+
+namespace ddwaf {
+namespace {
+// https://www.sqlite.org/lang_select.html
+// Identifiers: https://sqlite.org/lang_keywords.html
+re2::RE2 identifier_regex(
+    R"((?i)^(?:(?P<command>SELECT|DISTINCT|ALL|FROM|WHERE|GROUP|HAVING|WINDOW|VALUES|OFFSET|LIMIT|ORDER|BY|ASC|DESC|UNION|INTERSECT|EXCEPT|NULL|AS)|(?P<binary_operator>OR|AND|IN|BETWEEN|LIKE|GLOB|ESCAPE|COLLATE|REGEXP|MATCH|NOTNULL|ISNULL|NOT|IS)|(?P<identifier>[\x{0080}-\x{FFFF}a-zA-Z_][\x{0080}-\x{FFFF}a-zA-Z_0-9$]*|\$[0-9]+))(?:\b|\s|$))");
+
+} // namespace
+
+sqlite_tokenizer::sqlite_tokenizer(
+    std::string_view str, std::unordered_set<sql_token_type> skip_tokens)
+    : sql_tokenizer(str, std::move(skip_tokens))
+{
+    if (!identifier_regex.ok()) {
+        throw std::runtime_error(
+            "sqlite identifier regex not valid: " + identifier_regex.error_arg());
+    }
+}
+
+void sqlite_tokenizer::tokenize_command_operator_or_identifier()
+{
+    sql_token token;
+    token.index = index();
+
+    auto remaining_str = substr(index());
+
+    re2::StringPiece binary_op;
+    re2::StringPiece command;
+    re2::StringPiece ident;
+
+    const re2::StringPiece ref(remaining_str.data(), remaining_str.size());
+    if (re2::RE2::PartialMatch(ref, identifier_regex, &command, &binary_op, &ident)) {
+        // At least one of the strings will contain a match
+        if (!binary_op.empty()) {
+            token.type = sql_token_type::binary_operator;
+            token.str = substr(token.index, binary_op.size());
+            advance(token.str.size() - 1);
+        } else if (!command.empty()) {
+            token.type = sql_token_type::command;
+            token.str = substr(token.index, command.size());
+            advance(token.str.size() - 1);
+        } else if (!ident.empty()) {
+            token.type = sql_token_type::identifier;
+            token.str = substr(token.index, ident.size());
+            advance(token.str.size() - 1);
+        }
+        emplace_token(token);
+    }
+}
+
+void sqlite_tokenizer::tokenize_inline_comment_or_operator()
+{
+    // The first character is / so it can be a comment or a binary operator
+    sql_token token;
+    token.index = index();
+    if (advance() && peek() == '*') {
```

Comment:
Maybe for simplicity `advance() && peek()` could be behind a function called `next`

Reply:
I get your point, although the `next()` function exists and it's used as a `lookahead(1)` without advancing the cursor.